nixos/modules/services/misc/jellyfin.nix

```diff
@@ -0,0 +1,59 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.jellyfin;
+in
+{
+  options = {
+    services.jellyfin = {
+      enable = mkEnableOption "Jellyfin Media Server";
+
+      user = mkOption {
+        type = types.str;
+        default = "jellyfin";
+        description = "User account under which Jellyfin runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "jellyfin";
+        description = "Group under which jellyfin runs.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.jellyfin = {
+      description = "Jellyfin Media Server";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = rec {
+        Type = "simple";
+        User = cfg.user;
+        Group = cfg.group;
+        StateDirectory = "jellyfin";
+        CacheDirectory = "jellyfin";
+        ExecStart = "${pkgs.jellyfin}/bin/jellyfin --datadir '/var/lib/${StateDirectory}' --cachedir '/var/cache/${CacheDirectory}'";
+        Restart = "on-failure";
+      };
+    };
+
+    users.users = mkIf (cfg.user == "jellyfin") {
```

Comment:
Is there a specific reason that the user should be able to specify the user and group name that `jellyfish` will run as?

Reply:
Some users may want to change the user/group of the jellyfin service to be able to read media files from another user. I don't think it's the best way to do this, but it definitely is the simplest one.

Reply:
Here are the two use cases I can think of off the top of my head and how most distributions handle them:
- Media is stored under the programs data directory in which case each user wanting to access media has `jellyfish` added to their supplementary groups, and permissions set to read and write access for the group on this directory.
- Media is stored under users home directory in which case the `jellyfish` user has each user with media added to its supplementary groups.

Granted the second option doesn't handle the situation very well on multi user computers, but if either of those solutions don't fit the problem ACLs are an excellent solution.

To be clear: yes I know *every* other module has these options and so I'm happy to accept this module to do the same but I want to have people consider if the user and group options are actually necessary/will actually ever be used or if the option is added just because nix is so powerful and can easily do so. :smile: